Src/Support/IntegrationTests/PerCallAuthTests.cs

```diff
@@ -61,6 +64,151 @@ public void OverrideServiceCredentialWithPerCallCredential()
             Assert.NotNull(buckets.Items);
         }
 
+        private class RequestInterceptorCredentialWrapper : IConfigurableHttpClientInitializer, IHttpExecuteInterceptor
+        {
+            private AccessTokenCredential _credential;
+            public HttpRequestMessage Request { get; private set; }
+
+            public RequestInterceptorCredentialWrapper(string accessToken, IAccessMethod accessMethod) =>
+                _credential = new AccessTokenCredential(accessToken, accessMethod);
+
+            public void Initialize(ConfigurableHttpClient httpClient)
+            {
+                _credential.Initialize(httpClient);
+                httpClient.MessageHandler.AddExecuteInterceptor(this);
+            }
+
+            public Task InterceptAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                Request = request;
+                return Task.FromResult(true);
+            }
+        }
+
+        [Fact]
+        public void OverridesServiceCredentialWithPerCallCredential_QueryString()
+        {
+            var serviceCredential = new RequestInterceptorCredentialWrapper("SERVICE_CREDENTIAL_TOKEN", new BearerToken.QueryParameterAccessMethod());
+            var callCredential = new AccessTokenCredential("CALL_CREDENTIAL_TOKEN", new BearerToken.QueryParameterAccessMethod());
+
+            // Set one credential in the client
+            StorageService client = new StorageService(new BaseClientService.Initializer
+            {
+                ApplicationName = "IntegrationTest",
+                HttpClientInitializer = serviceCredential
+            }); ;
+
+            // Create a request with a per-call credential.; this overrides the service-level credential.
+            var request = client.Buckets.List(Helper.GetProjectId()).AddCredential(callCredential);
+
+            // This should throw unauthorized, the credentials are fake, but let's make sure the
+            // request got to the server so we know all appropiate interceptors executed.
+            var exception = Assert.ThrowsAny<GoogleApiException>(() => request.Execute());
+            Assert.Equal(401, exception.Error.Code);
+
+            string query = serviceCredential.Request.RequestUri.Query;
+
+            int serviceIndex = query.IndexOf("access_token=SERVICE_CREDENTIAL_TOKEN", StringComparison.OrdinalIgnoreCase);
+            int callIndex = query.LastIndexOf("access_token=CALL_CREDENTIAL_TOKEN", StringComparison.OrdinalIgnoreCase);
```

Comment:
Why is this LastIndexOf where before we had IndexOf? Would it make sense to use Contains instead?